lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1474,6 +1491,91 @@ spec = do
               txs2 <- listTransactions ctx w (Just te) (Just te) Nothing
               length <$> [txs1, txs2] `shouldSatisfy` all (== 0)
 
+    it "TRANS_DELETE_01 - Single Output Transaction" $ \ctx -> do
+        (wa, wb) <- (,) <$> fixtureWallet ctx <*> fixtureWallet ctx
+        addrs <- listAddresses ctx wb
+
+        let amt = 1
+        let destination = (addrs !! 1) ^. #id
+        let payload = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt},
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+        let (feeMin, feeMax) = ctx ^. feeEstimator $ TxDescription
+                { nInputs = 1
+                , nOutputs = 1
+                }
+
+        r <- request @(ApiTransaction t) ctx (postTxEp wa) Default payload
+        verify r
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectFieldBetween DSL.amount (feeMin + amt, feeMax + amt)
+            , expectFieldEqual DSL.direction Outgoing
+            , expectFieldEqual DSL.status Pending
+            ]
+
+        let (ApiTransaction txId _ _ _ _ _ _ _ _) = getFromResponse Prelude.id r
+
+        ra <- request @ApiWallet ctx (getWalletEp wa) Default Empty
+        verify ra
+            [ expectSuccess
+            , expectFieldBetween balanceTotal
+                ( faucetAmt - feeMax - amt
+                , faucetAmt - feeMin - amt
+                )
+            , expectFieldEqual balanceAvailable (faucetAmt - faucetUtxoAmt)
+            ]
+
+        rDel <- request @ApiTxId ctx (deleteTxEp wa (ApiTxId txId)) Default Empty
```

Comment:
There's a race condition here, although it's _probably_ fine in most cases, but the transaction might have already been accepted at this point. I don't think there's much we can do apart from retrying making another transaction and hoping it works. 

There's a `eventually` combinator in the DSL that could be quite handy here to wrap around both the creation of the transaction and this delete call.

Reply:
This still needs to be addressed